http.go

```diff
@@ -92,7 +103,7 @@ func (client *Client) Request(rawPayload []byte, method, route string, query url
 				return nil, nil, err
 			}
 		}
-		resp, err := (&http.Client{}).Do(callSummary.HTTPRequest)
+		resp, err := DefaultHTTPClient.Do(callSummary.HTTPRequest)
```

Comment:
I like this, but let's it would be better to have this as a property of `*tcclient.Client`, rather than a global package variable. E.g.

```
type Client struct {
  ....
  HTTPClient ReducedHTTPClient
}
```

Since this interface only has one method, I think defining `ReducedHTTPClient` as a _function_ rather than an interface would be even better, i.e. instead of above, have:

```
type ReducedHTTPClient func(req *http.Request) (*http.Response, error)
```

For example, an advantage of this is that the passed-in method does not need to be called `Do`, but any suitable function can be passed in which satisfies the function signature.

Then we can remove the `DefaultHTTPClient` from the package. Instead, we would probably in-line the check:

```
var resp *http.Response
if c := client.HTTPClient; c != nil {
    resp, err = c(callSummary.HTTPReqest)
} else {
    resp, err = &http.Client{}.Do(callSummary.HTTPRequest)
}
```

The reason to inline the check is to account for the HTTP client not being set, which could also be done by making it not possible to instantiate `tcclient.Client` directly, but force using a constructor that guarantees setting the internal http client function - however, I think this is overkill, and it is nicer being able to use a struct literal for `tcclient.Client` and the `Request` method to be able to account for it.